packages/http-server-csharp/src/lib/service.ts

```diff
@@ -866,51 +994,58 @@ export async function $onEmit(context: EmitContext<CSharpServiceEmitterOptions>)
       };
     }
     #emitOperationResponses(operation: HttpOperation): EmitterOutput<string> {
+      function isValid(program: Program, response: HttpOperationResponse) {
+        return (
+          !isErrorModel(program, response.type) &&
+          getCSharpStatusCode(response.statusCodes) !== undefined
+        );
+      }
       const builder: StringBuilder = new StringBuilder();
       let i = 0;
       const validResponses = operation.responses.filter(
         (r) =>
           !isErrorModel(this.emitter.getProgram(), r.type) &&
           getCSharpStatusCode(r.statusCodes) !== undefined,
       );
-      for (const response of validResponses) {
-        i++;
-        builder.push(code`${this.#emitOperationResponseDecorator(response)}`);
-        if (i < validResponses.length) {
-          builder.pushLiteralSegment("\n");
-        }
-      }
-
       for (const response of operation.responses) {
-        if (!isEmptyResponseModel(this.emitter.getProgram(), response.type))
-          this.emitter.emitType(response.type);
+        const [responseType, resolvedResponse] = this.#resolveOperationResponse(
+          response,
+          operation.operation,
+        );
+        if (isValid(this.emitter.getProgram(), response)) {
+          i++;
+          builder.push(
+            code`${this.#emitOperationResponseDecorator(response, resolvedResponse, responseType)}`,
+          );
+          if (i < validResponses.length) {
```

Comment:
I think this code is the only reason we have validResponses in here, if that is the case, probably it would be simpler to do the `builder.pushLiteralSegment("\n"); ` when there is something already in `builder`, like doing the check before doing the ` builder.push(code....);` and if there is something in builder already add the `\n`, in that case we don't need `validResponses`